specs/core/1_shard-data-chains.md

```diff
@@ -222,3 +222,243 @@ The `shard_chain_commitment` is only valid if it equals `compute_commitment(head
+                          signature=response.signature,
+                          domain=get_domain(state, get_current_epoch(state), DOMAIN_CUSTODY_INTERACTIVE))
+    # Update challenge data
+    challenge_data.deadline=FAR_FUTURE_EPOCH
+    validator.withdrawable_epoch = get_current_epoch(state) + MAX_POC_RESPONSE_DEPTH
+```
+
+Once a response provides 32 hashes, the challenger has the right to choose any one of them that they feel is constructed incorrectly to continue the game. Note that eventually, the game will get to the point where the `new_custody_tree_node` is a leaf node. We define an `InteractiveCustodyChallengeContinuation` object as follows:
+
+```python
+{
+    'challenger_index: 'uint64',
+    'responder_index': 'uint64',
+    'sub_index': 'uint64',
+    'new_custody_tree_node': 'bytes32',
+    'proof': ['bytes32'],
+    'signature': 'bytes96'
+}
+```
+
+Here's the function for verifying and processing a continuation challenge:
+
+```python
+def process_continuation(continuation: InteractiveCustodyChallengeContinuation,
+                         state: State):
+    responder = state.validator_registry[continuation.responder_index]
+    challenge_data = responder.interactive_custody_challenge_data
+    expected_depth = min(challenge_data.max_depth - challenge_data.depth, MAX_POC_RESPONSE_DEPTH)
+    # Verify we're not too late
+    assert get_current_epoch(state) < responder.withdrawable_epoch
+    # Verify the Merkle branch (the previous custody response provided the next level of hashes so the
+    # challenger has the info to make any Merkle branch)
+    assert verify_merkle_branch(
+        leaf=new_custody_tree_node,
+        branch=continuation.proof,
+        depth=expected_depth,
+        index=sub_index,
+        root=challenge_data.current_custody_tree_node
+    )
+    # Verify signature
+    assert bls_verify(message_hash=signed_root(continutation, 'signature'),
+                      pubkey=responder.pubkey,
+                      signature=continutation.signature,
+                      domain=get_domain(state, get_current_epoch(state), DOMAIN_CUSTODY_INTERACTIVE))
+    # Update the challenge data
+    challenge_data.current_custody_tree_node = continuation.new_custody_tree_node
+    challenge_data.depth += expected_depth
+    challenge_data.deadline = get_current_epoch(state) + MAX_POC_RESPONSE_DEPTH
+    validator.withdrawable_epoch = FAR_FUTURE_EPOCH
+    challenge_data.offset = challenger_data.offset * 2**expected_depth + sub_index
+```
+
+Once the `new_custody_tree_node` reaches the leaves of the tree, the responder can no longer provide a valid `InteractiveCustodyChallengeResponse`; instead, the responder or the challenger must provide a branch response that provides a branch of the original data tree, at which point the custody leaf equation can be checked and either side of the custody game can "conclusively win".
+
+```python
+def process_branch_response(response: BranchResponse,
```

Comment:
`BranchResponse` is another tx that can be included in the block? Currently undefined.

Reply:
It's defined in #587 :laughing: 

In reality the branch response processing here and there should be a single unified function, but I can't really do that now because #587 is not merged yet.

The idea is that we can dual-use the `BranchResponse` object for responding to regular branch challenges and the last stage of this interactive game.